node/podcontroller.go

```diff
@@ -67,12 +77,22 @@ type PodLifecycleHandler interface {
 	// concurrently outside of the calling goroutine. Therefore it is recommended
 	// to return a version after DeepCopy.
 	GetPodStatus(ctx context.Context, namespace, name string) (*corev1.PodStatus, error)
+}
 
-	// GetPods retrieves a list of all pods running on the provider (can be cached).
-	// The Pods returned are expected to be immutable, and may be accessed
+type PodUIDLifecycleHandler interface {
+	podLifecycleHandler
+
+	// GetPodByUID retrieves a pod by name and UID from the provider (can be cached).
+	// The Pod returned is expected to be immutable, and may be accessed
 	// concurrently outside of the calling goroutine. Therefore it is recommended
 	// to return a version after DeepCopy.
-	GetPods(context.Context) ([]*corev1.Pod, error)
+	GetPodByUID(ctx context.Context, namespace, name string, uid types.UID) (*corev1.Pod, error)
```

Comment:
why don't we just add these to the PodLifecycleHandler directly? or do you want some type safety against calling GetPods/GetPodStatus?

Reply:
For the type safety. I want this to be backwards-compatible (I'm going to see if we can get it upstreamed), but I also don't want the old methods to be accidentally called internally.